pkg/udev/common.go

```diff
@@ -143,9 +146,16 @@ func (device *UdevDevice) GetDevLinks() map[string][]string {
 	byIdLink := make([]string, 0)
 	byPathLink := make([]string, 0)
 	for _, link := range strings.Split(device.GetPropertyValue(UDEV_DEVLINKS), " ") {
+		// devlink is like - /dev/disk/by-id/scsi-0Google_PersistentDisk_demo-disk
+		// parts = ["", "dev", "disk", "by-id", "scsi-0Google_PersistentDisk_demo-disk"]
+		// parts[4] containis link index like model or wwn or sysPath (wwn-0x5000c5009e3a8d2b) (ata-ST500LM021-1KJ152_W6HFGR)
 		parts := strings.Split(link, "/")
 		if util.Contains(parts, BY_ID_LINK) {
-			byIdLink = append(byIdLink, link)
+			if strings.HasPrefix(parts[LINK_ID_INDEX], device.GetPropertyValue(UDEV_BUS)) && strings.HasSuffix(parts[LINK_ID_INDEX], device.GetPropertyValue(UDEV_SERIAL_FULL)) {
```

Comment:
Write a comment explaining the logic of promoting the particular by-id entry at the top of the array.